app/models/ZkKafka.scala

```diff
@@ -40,20 +40,26 @@ object ZkKafka {
   def getTopologies: Seq[Topology] = {
     zkClient.getChildren.forPath(makePath(Seq(stormZkRoot))).asScala.map({ r =>
       getSpoutTopology(r)
-    }).sortWith(topoCompFn)
+    }).flatten.sortWith(topoCompFn)
   }
 
-  def getSpoutTopology(root: String): Topology = {
+  def getZkData(path: String): Option[Array[Byte]] = Option(zkClient.getData.forPath(path))
+
+  def getSpoutTopology(root: String): Option[Topology] = {
     // Fetch the spout root
     val s = zkClient.getChildren.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root)))))
     // Fetch the partitions so we can pick the first one
     val parts = zkClient.getChildren.forPath(makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(s.get(0)))))
```

Comment:
While we're here and working on this, the `.get(0)` in there are unsafe. It should probably map off the `s`, creating a nested function for the stuff below…